internal/cache/metadata/stat_cache.go

```diff
@@ -247,8 +247,19 @@ func (sc *statCacheBucketView) LookUpFolder(
 	now time.Time) (bool, *gcs.Folder) {
 	// Look up in the LRU cache.
 	hit, entry := sc.sharedCacheLookup(folderName, now)
+
 	if hit {
-		return hit, entry.f
+		// Adds negative entry scenario
+		if entry.f == nil && entry.m == nil {
+			return true, nil
+		}
+
+		// Adds uninitiated object scenario
```

Comment:
What is uninitiated object scenario? Could we add clarity here.

Reply:
Updated comment to be more descriptive, also corresponding test is also updated test name (Test_ShouldReturnHitFalseWhenOnlyObjectAlreadyHasEntry) for better documentation